### PR TITLE
interfaces: add system-source-code for access to /usr/src

### PR DESCRIPTION
**Pull request by jdstrand**
I went back and forth on the name before finally landing on system-source-code. Thoughts:

* usr-src: this is accurate and understandable, but leaves no room for extending the interface
* system-usr-src: same as above
* system-source-code: this seems to convey the best balance of what it is, how the FHS defines it and allows for future additions

Adding the 'Needs Samuele review' to finalize the name.

Note, /usr/src is already handled by snap-confine with accesses already allowed via `system-trace` so mount rules are not needed.

References:
* https://forum.snapcraft.io/t/request-for-classic-confinement-device-tree-compiler/17490/13
* https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.html#usrsrcSourceCode

### CODE REVIEW

**Comment by zyga** [comment]
Added Graham as assignee so that new interfaces are included in the documentation stack.

**Comment by jdstrand** [comment]
> Technically LGTM, nothing looks wrong here.
> 
> Did you consider making use of the mount profiles and removing the hard-coded mount from snap-confine?

I did not. AFAIR, we do not have a clear path forward for when to remove things from snap-confine. If this is one of those times, I can do it.

File: interfaces/builtin/system_source_code.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.html#usrsrcSourceCode
+const systemSourceCodeSummary = `allows read-only access to /usr/src of the system`
```
Comment (pedronis) [inline interfaces/builtin/system_source_code.go:23]: is this really meant to be "of" ? or "on" ? I would have expected either:

* access to the /usr/src of the system
* access to /usr/src on the system

Comment (degville) [reply interfaces/builtin/system_source_code.go:23]: fwiw, I think it should be `on the system`.

Comment (pedronis) [reply interfaces/builtin/system_source_code.go]: changed to `on the system`

Comment (jdstrand) [reply interfaces/builtin/system_source_code.go]: Sorry, yes, "on the system" is good. Thanks for updating it.

**Pull request merged**